sympy/core/tests/test_diff.py

```diff
@@ -135,7 +135,9 @@ def test_diff_nth_derivative():
         Eq(y, 0) & Eq(Max(0, -y + n), 0)),
         (2*factorial(n)/(factorial(y)*factorial(-y + n)), Eq(y, 0) & Eq(Max(0,
         -y + n), 1)), (0, True)), (y, 0, n)))
-    # TODO: assert diff(x**2, (x, n)) == x**(2-n)*ff(2, n)
+    #Check the nth derivative of x**2
+    assert diff(x**2, (x, n))== Derivative(x*x, (x, n))
```

Comment:
The reason that this is a TODO is not because the test needs to be added but because the functionality needs to be added for computing the nth derivative.

Reply:
This test isn't really worth adding so I think it's better to leave the code as it is and close this PR.

Reply:
Oh! pardon i am new to sympy and open source in genral so didn't knew this. Thanks @oscarbenjamin for pointing it out i got confused by the TODO  being mentioned in the test_diff file.
Moreover if the functionality is needed to be added i would like to work on this , can you help me out by guiding how should  i implement this.

Reply:
You'd have to look into the `Pow._eval_derivative_ntimes` method I think.